### ISSUE TITLE
Pokemon R/S link error

### ISSUE THREAD

**Issue by Fortranm** [alyosha-tas/GBAHawk#16]
![image](https://github.com/alyosha-tas/GBAHawk/assets/19356108/9c135c22-476e-4841-bf94-1c82eb42f9b8)
This happened when I tried to trade between Ruby and Sapphire.

Pokemon - Ruby Version (USA)
SHA1:F28B6FFC97847E94A6C21A63CACF633EE5C8DF1E
MD5:53D1A2027AB49DF34A689FAA1FB14726

Pokemon - Sapphire Version (USA, Europe) (Rev 2)
SHA1:89B45FB172E6B55D51FC0E61989775187F6FE63C
MD5:9BC2B765CA6997175FAC51E6CDC29089

Movie file: [PokeRS.zip](https://github.com/alyosha-tas/GBAHawk/files/12256510/PokeRS.zip)

**Comment by alyosha-tas** [comment]
I'll need to test this on the debug core. Can you send a movie that runs from power on?

**Comment by alyosha-tas** [comment]
I finally figured out what was wrong here. Turns out its actually an IRQ issue, a serial IRQ was being enabled exactly when IRQ master enable was happening and missed the check. Quite miraculous it happened on exactly the needed cycle. Missing an IRQ through off the count the consoles were keeping, causing the link error.

This will eventually be fixed with the other open IRQ issue.

In the mean time if you just start the transfer on a different frame you have a good chance of missing the bug.
